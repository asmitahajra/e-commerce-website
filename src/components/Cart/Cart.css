.cart-page {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.cart-page> h1 {
    font-size: large;
    font-weight: bolder;
}

.checkout-continue-container {
    display: flex;
    justify-content: space-between;
    padding: 0.5rem;
}

.continue-shopping {
    height: 2rem;
    width: 10rem;
    border: 2rem;
    box-shadow: 0.5rem 0.5rem 0.5rem #888888;
}

.checkout-container {
    padding: 0.5rem;
    width: 15rem;
    height: 10rem;
    border: 1px solid black;
    display: flex;
    flex-direction: column;
}

.total-value >h2 {
    font-size: medium;
    font-weight: normal;
}

.checkout-container .checkout{
    height: 3rem;
    background-color: khaki;
    width: 10rem;
    border: 2rem;
    box-shadow: 0.5rem 0.5rem 0.5rem #888888;
}