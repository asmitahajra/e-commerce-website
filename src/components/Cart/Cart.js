import React from 'react';
import './Cart.css';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import CartTable from '../CartTable/CartTable';

const Cart = ({ cartItems }) => {
  const totalValue = cartItems.reduce(
    (cartTotal, currentItem) => cartTotal + currentItem.count * currentItem.price, 0,
  );
  const itemNo = cartItems.length;
  // const myJSON = JSON.stringify(cartItems);
  // console.log(JSON.stringify(cartItems));
  return (
    <div className="cart-page">
      <div className="cart-table-body">
        <h1>
          Your Basket (
          {itemNo}
          {' '}
          items)
        </h1>
        <hr />
        <div className="tableItems">
          <CartTable cartItems={cartItems} />
        </div>
      </div>
      <div className="checkout-continue-container">
        <Link to="/"><button type="button" className="continue-shopping">Continue Shopping</button></Link>
        <div className="checkout-container">
          <div className="total-value">
            <h2>
              TOTAL
              {' '}
              Rs.
              {totalValue}
            </h2>
            <hr />
          </div>
          <Link to="/checkout"><button className="checkout" type="button">checkout</button></Link>
        </div>
      </div>
    </div>
  );
};

Cart.propTypes = {
  cartItems: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    company: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    count: PropTypes.number.isRequired,
  })).isRequired,
};

export default Cart;
