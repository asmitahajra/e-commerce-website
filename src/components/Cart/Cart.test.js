/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Cart from './Cart';

describe(Cart.name, () => {
  const mockProp = {
    cartItems: [{
      id: 1,
      name: 'apple',
      price: 20,
      count: 1,
      image: 'apple.jpg',
      company: 'FRSHO',
    }, {
      id: 2,
      name: 'banana',
      price: 30,
      count: 1,
      image: 'banana.jpg',
      company: 'FRSHO',
    }],
  };

  afterEach(() => jest.clearAllMocks());

  test('should match snapshot', () => {
    const { container } = render(<BrowserRouter><Cart {...mockProp} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });

  test('should display Your Basket and no. of items', () => {
    render(<BrowserRouter><Cart {...mockProp} /></BrowserRouter>);
    screen.getByText('Your Basket (2 items)');
  });

  test('should render CartTable', () => {
    render(<BrowserRouter><Cart {...mockProp} /></BrowserRouter>);
    screen.getByTestId('cart-table');
  });

  test('should open Homepage when continue shopping clicked', () => {
    render(<BrowserRouter><Cart {...mockProp} /></BrowserRouter>);
    const continueShoppingButton = screen.getByText('Continue Shopping');
    fireEvent.click(continueShoppingButton);
    expect(document.location.href).toBe('http://localhost/');
  });

  test('should display total amount', () => {
    render(<BrowserRouter><Cart {...mockProp} /></BrowserRouter>);
    screen.getByText('TOTAL Rs.50');
  });

  test('should open checkout form when checkout clicked', () => {
    render(<BrowserRouter><Cart {...mockProp} /></BrowserRouter>);
    const checkoutButton = screen.getByText('checkout');
    fireEvent.click(checkoutButton);
    expect(document.location.href).toBe('http://localhost/checkout');
  });
});
