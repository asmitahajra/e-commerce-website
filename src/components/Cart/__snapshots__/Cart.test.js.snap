// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should match snapshot 1`] = `
<div>
  <div
    class="cart-page"
  >
    <div
      class="cart-table-body"
    >
      <h1>
        Your Basket (
        2
         
        items)
      </h1>
      <hr />
      <div
        class="tableItems"
      >
        <div
          data-testid="cart-table"
        >
          <div
            class="cart-table-container"
          >
            <table
              class="cart-table"
            >
              <thead
                class="table-header"
              >
                <tr
                  class=""
                >
                  <th>
                    ITEM DESCRIPTION
                  </th>
                  <th>
                    UNIT PRICE
                  </th>
                  <th>
                    QUANTITY
                  </th>
                  <th>
                    SUBTOTAL
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    class="category-header"
                    colspan="4"
                  >
                    Fruits and Vegetables
                  </td>
                </tr>
                <tr>
                  <td>
                    apple
                  </td>
                  <td>
                    20
                  </td>
                  <td>
                    1
                  </td>
                  <td>
                    20
                  </td>
                </tr>
                <tr>
                  <td>
                    banana
                  </td>
                  <td>
                    30
                  </td>
                  <td>
                    1
                  </td>
                  <td>
                    30
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <div
      class="checkout-continue-container"
    >
      <a
        href="/"
      >
        <button
          class="continue-shopping"
          type="button"
        >
          Continue Shopping
        </button>
      </a>
      <div
        class="checkout-container"
      >
        <div
          class="total-value"
        >
          <h2>
            TOTAL
             
            Rs.
            50
          </h2>
          <hr />
        </div>
        <a
          href="/checkout"
        >
          <button
            class="checkout"
            type="button"
          >
            checkout
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
